frontend/src/components/Locations/LocationForm.tsx

```diff
@@ -1,222 +1,309 @@
-    pickupOrDelivery: number,
-    id: String,
-    date: String,
-    driver: String,
-    vehicle: String,
-    name: String,
-    EntityType: String,
-    LocationType: String,
-    Demographic: String,
-    FoodType: String,
-    Area: String,
-    lbsDroppedOff: number
+  pickupOrDelivery: number;
+  id: string;
+  date: string;
+  driver: string;
+  vehicle: string;
+  name: string;
+  EntityType: string;
+  LocationType: string;
+  Demographic: string;
+  FoodType: string;
+  Area: string;
+  lbsDroppedOff: number;
 }
 
-
-interface Props{
-  current                 : locale,
-  createNew               : boolean,
-  setLocation             : Function,
-  PickupDeliveryObject    : pickupDeliveryObjectSchema,
-  setPickupDeliveryObject : Function,
-  setForceNext            : Function
+interface Props {
+  current: locale;
+  createNew: boolean;
+  setLocation: Function;
+  PickupDeliveryObject: pickupDeliveryObjectSchema;
+  setPickupDeliveryObject: Function;
+  setForceNext: Function;
 }
 
-const LocationForm = ({current, createNew, setLocation, PickupDeliveryObject, setPickupDeliveryObject, setForceNext} : Props) =>{
-  const [active, setActive] = useState(""); // State for radio buttons
-  const [isClicked, setIsClicked] = useState(true); // State for radio buttons
+const LocationForm = ({
+  current,
+  createNew,
+  setLocation,
+  PickupDeliveryObject,
+  setPickupDeliveryObject,
+  setForceNext
+}: Props) => {
+  const [active, setActive] = useState<string[]>([]);
+  const [isValid, setIsValid] = useState(true); // State for checkbox buttons
+  const [isOtherClicked, setOtherClicked] = useState(false);
+  const [isProduceClicked, setProduceClicked] = useState(false);
+  const [currentFoodType, setCurrentFoodType] = useState('');
+  // State for checkbox buttons
 
   // State for if user is adding a new location
-  const [donorName, setName] = useState("");
-  const [donorLocationType, setDonorLocationType] = useState("");
-  const [donorEntityType, setDonorEntityType] = useState("");
-  const [food, setFoodType] = useState("");
-  const [demographic, setDemographic] = useState("");
-  const [area, setArea] = useState("");
+  const [donorName, setName] = useState('');
+  const [donorLocationType, setDonorLocationType] = useState('');
+  const [donorEntityType, setDonorEntityType] = useState('');
+  const [food, setFoodType] = useState('');
+  const [demographic, setDemographic] = useState('');
+  const [area, setArea] = useState('');
+  const [input, setInput] = useState('');
+
+  useEffect(() => {
+    console.log(isOtherClicked);
+  }, [isOtherClicked]);
+
+  useEffect(() => {
+    console.log(isProduceClicked);
+  }, [isProduceClicked]);
 
   const { name } = current;
 
   const submitPressed = () => {
-    if (createNew === false && active !== ""){
+    if (isOtherClicked) {
+      if (input.length != 0 && input != ' ') {
+        active.push(input);
+        setActive(active);
+      }
+    }
+
+    if (isProduceClicked) {
```

Comment:
Not all of the entities have produce and some of them will have multiple food types. In order to make this a more generalized solution, I think we need to modify the logic surrounding the checks. There are a lot of really good solutions online for keeping track of active elements in a checked list when there are multiple. 

Here's one: https://contactmentor.com/checkbox-list-react-js-example/

To be clear right now the checkboxes are being generated by `current.FoodType` which is a single string. On other branches this has been changed to a list. So you can keep creating checkboxes the way you're currently doing but just modify the logic for selecting boxes to be generalized for multiple checks.